src/winscard_msg_srv.c

```diff
@@ -157,9 +161,18 @@ INTERNAL int32_t InitializeSocket(void)
 	}
 
 	/*
-	 * Chmod the public entry channel
+	 * Chmod the public entry channel securely
 	 */
-	(void)chmod(PCSCLITE_CSOCK_NAME, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH);
+	if ((fd = open(PCSCLITE_CSOCK_NAME, O_RDWR)) == -1)
+	{
+		Log2(PCSC_LOG_CRITICAL, "Cannot open %s",
+			 strerror(errno));
+		return -1;
+	}
+	
+	(void)fchmod(fd, mode);
+
+	(void)close(fd);
 
```

Comment:
Why is it better than a call to chmod()?

Reply:
Reviewed again. This change is not necessary.

-  In case when binding socket, there exists symlink PCSCLITE_CSOCK_NAME, an error should be returned by function bind. 
- Another possibility for an attack is remove file PCSCLITE_CSOCK_NAME after calling listen() and than create symlink PCSCLITE_CSOCK_NAME, however this shouldn't be done because this file cannot be removed by an user who is not owner in that moment, am I right? If not this is a possible scenario for an attack.